src/engine/Match_search_mode.ml

```diff
@@ -579,6 +579,7 @@ let rec filter_ranges (env : env) (xs : (RM.t * MV.bindings list) list)
              let mvalue_to_expr m =
                match Metavariable.mvalue_to_any m with
                | G.E e -> Some e
+               | G.Name n -> Some (N n |> G.e)
```

Comment:
This can be tested independently, but it can only be reproduced with the Pro engine, so I don't think we should add a separate test for it. This PR fixes the following error:
```
!!!This is a proprietary extension of semgrep.!!!
!!!You should not call directly this program.!!!
WARNING: some files were skipped or only partially analyzed:
temp/p.py:1:0: "Internal matching error": An error occurred while invoking the Semgrep engine. Please help us fix this by creating an issue at https://github.com/returntocorp/semgrep

couldn't find metavar $MT in the match results.
```

Reply:
We can only get a `Name` in there using the Pro engine? Maybe it's worth a comment in the code explaining how do we get a `Name` there.

Reply:
When we capture `IdQualified` with `Id` metavariable, `G.Name` is generated. The Pro engine provides better name resolution, so `Id` global variables are promoted to `IdQualified` with the file path prefix.  Added the comment.

Reply:
Are you sure that semgrep/semgrep-proprietary#1450 is testing this change? I tried semgrep/semgrep-proprietary/pull/1450 without this change and it works fine. How to reproduce?

Reply:
did you test with `semgrep-core-proprietary` or `make test`? when you test with `semgrep-core-proprietary`, you need to add `-debug` to see the error message. For me, `make test` failed without this PR:

```
 ASSERT tests/interfile/naming/python/class_initializer/class_init.py:1:0: "Internal matching error": An error occurred while invoking the Semgrep engine. Please help us fix this by creating an issue at https://github.com/returntocorp/semgrep
 
 couldn't find metavar $MT in the match results.
 tests/interfile/naming/python/class_initializer/class_init.py:1:0: "Internal matching error": An error occurred while invoking the Semgrep engine. Please help us fix this by creating an issue at https://github.com/returntocorp/semgrep
 
 couldn't find metavar $MT in the match results.
 tests/interfile/naming/python/class_initializer/class_init.py:1:0: "Internal matching error": An error occurred while invoking the Semgrep engine. Please help us fix this by creating an issue at https://github.com/returntocorp/semgrep
 
 couldn't find metavar $MT in the match results.
 FAIL: The test raised an exception: Alcotest assertion failure
 FAIL tests/interfile/naming/python/class_initializer/class_init.py:1:0: "Internal matching error": An error occurred while invoking the Semgrep engine. Please help us fix this by creating an issue at https://github.com/returntocorp/semgrep
 
 couldn't find metavar $MT in the match results.
 tests/interfile/naming/python/class_initializer/class_init.py:1:0: "Internal matching error": An error occurred while invoking the Semgrep engine. Please help us fix this by creating an issue at https://github.com/returntocorp/semgrep
 
 couldn't find metavar $MT in the match results.
 tests/interfile/naming/python/class_initializer/class_init.py:1:0: "Internal matching error": An error occurred while invoking the Semgrep engine. Please help us fix this by creating an issue at https://github.com/returntocorp/semgrep
 
 couldn't find metavar $MT in the match results.
 
 Raised at Alcotest_engine__Test.check_err in file "src/alcotest-engine/test.ml", line 157, characters 20-48
 Called from Testo__Monad.catch in file "semgrep/libs/testo/core/Monad.ml", line 14, characters 6-13
```